### PR TITLE
Validation test harness

### PR DESCRIPTION
**Pull request by mvs5465**
First pass attempt at a test harness.

Run it as a python script: 
```
python main.py -f bsmLogDuringEvent.ini
```

It will use the payloadType field to determine whether or not to check the `bsmSource` field, other fields are universally checked.

It creates a Kafka Consumer with a 7 second timeout, validation is considered complete when the script hasn't detected any new messages for 7 seconds.

### CODE REVIEW

File: validation-script/main.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    print("[INFO] Running self-test of validation routine...")
+    MetadataValidator.validate_metadata(json.loads(self_test_data)['metadata'])
+    print("[INFO] Self test passed.")
+
+# main
+def main():
+    parser = ArgumentParser()
+    parser.add_argument("-f", "--file", dest="filepath", help="Path to file used for testing.", metavar="FILE", required=True)
```
Comment (hmusavi) [inline validation-script/main.py]: I think we're gonna need to have lots of config items and best way to pass them on is via a config file. Let's read a config file that could be either standard `.ini` file which python has built-in support for I think, or make it json format which we will be able to parse and pick from. File name could be defined in the config file.

File: validation-script/main.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    args = parser.parse_args()
+
+    # local test
+    # self_test()
+
+    # Create a kafka consumer and wait for it to connect
+    print("[INFO] Preparing Kafka listener...")
+    topic_name='topic.OdeBsmJson'
```
Comment (hmusavi) [inline validation-script/main.py]: topic name should also be in the config file.

File: validation-script/validator.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+# serialId
+#   streamId
+#   bundleSize
+#   bundleId
+#   recordId
+#   serialNumber
+# (not included) validSignature
+##################################
+
+# Value constraints
```
Comment (hmusavi) [inline validation-script/validator.py]: All these should be in the config file or a separate expected output file.

File: validation-script/validator.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+BUNDLE_ID_UPPER_LIMIT = 9223372036854775807
+BUNDLE_ID_LOWER_LIMIT = 0
+RECORD_ID_UPPER_LIMIT = 2147483647
+RECORD_ID_LOWER_LIMIT = 0
+SERIAL_NUMBER_UPPER_LIMIT = 9223372036854775807
+SERIAL_NUMBER_LOWER_LIMIT = 0
+
+# validates all fields of the metadata
+def validate_metadata(metadata, errored):
+    validate_field(errored, 'payloadType', metadata.get('payloadType'))
```
Comment (hmusavi) [inline validation-script/validator.py]: All these calls to `validate_medatdata` should ideally be made in a loop. The list that is looped through should come from the config file or from the combination of config, input file and expected output file.

File: validation-script/bsmLogDuringEvent.ini [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+[bsmSource]
+Path = metadata.bsmSource
+Type = enum
+Values = ["RV", "EV", "unknown"]
+
+[payloadType]
+Path = metadata.payloadType
+Type = string
+EqualsValue = us.dot.its.jpo.ode.model.OdeBsmPayload
```
Comment (mestifanos) [inline validation-script/bsmLogDuringEvent.ini]: I was not sure if the EqualsValue need to be assigned to a string.  In the Excel spreadsheet does not have a default value.

**Pull request merged**